packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -620,6 +620,28 @@ module A = {
       | Some(o) => o
       | None => []
       }
+    let rec arrSomeToSomeArr = (optionals: array<option<'a>>): option<array<'a>> => {
```

Comment:
does this return None when any of the elements are None, or when there are no non-None elements? (At least add a comment)

Reply:
I also suspect you could do this in a simpler way without recursion. See Belts array methods. https://rescript-lang.org/docs/manual/latest/api/belt/array